File: View/Components/StandingListViewItem.vb
Imports POFF.Kicker.Model

Namespace Components

    Public Class StandingListViewItem
        Inherits ListViewItem

        Public Sub New(standing As Standing)
            MyBase.New(standing.Place.ToString)
            With standing
                Me.SubItems.Add(.Team.Name)
                Me.SubItems.Add(.Points.ToString)
                Me.SubItems.Add(.WonSetCount.ToString)
                Me.SubItems.Add(String.Format("{0}:{1}", .Goals, .GoalsAgainst))
                Me.SubItems.Add(.MatchCount.ToString)
            End With

            StandingValue = standing
        End Sub

        Private ReadOnly StandingValue As Standing
        Public ReadOnly Property Standing() As Standing
            Get
                Return StandingValue
            End Get
        End Property

    End Class

End Namespace

File: View/Components/StandingListViewItem.cs
using System.Windows.Forms;
using POFF.Kicker.Model;

namespace POFF.Kicker.View.Components
{

    public class StandingListViewItem : ListViewItem
    {

        public StandingListViewItem(Standing standing) : base(standing.Place.ToString())
        {
            SubItems.Add(standing.Team.Name);
            SubItems.Add(standing.Points.ToString());
            SubItems.Add(standing.WonSetCount.ToString());
            SubItems.Add(string.Format("{0}:{1}", standing.Goals, standing.GoalsAgainst));
            SubItems.Add(standing.MatchCount.ToString());

            StandingValue = standing;
        }

        private readonly Standing StandingValue;
        public Standing Standing
        {
            get
            {
                return StandingValue;
            }
        }

    }

}